% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
\documentclass[
]{article}
\usepackage{xcolor}
\usepackage[margin=1in]{geometry}
\usepackage{amsmath,amssymb}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math} % this also loads fontspec
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
\usepackage{lmodern}
\ifPDFTeX\else
  % xetex/luatex font selection
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{graphicx}
\makeatletter
\newsavebox\pandoc@box
\newcommand*\pandocbounded[1]{% scales image to fit in text height/width
  \sbox\pandoc@box{#1}%
  \Gscale@div\@tempa{\textheight}{\dimexpr\ht\pandoc@box+\dp\pandoc@box\relax}%
  \Gscale@div\@tempb{\linewidth}{\wd\pandoc@box}%
  \ifdim\@tempb\p@<\@tempa\p@\let\@tempa\@tempb\fi% select the smaller of both
  \ifdim\@tempa\p@<\p@\scalebox{\@tempa}{\usebox\pandoc@box}%
  \else\usebox{\pandoc@box}%
  \fi%
}
% Set default figure placement to htbp
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\usepackage{bookmark}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\urlstyle{same}
\hypersetup{
  pdftitle={GSE211398\_race\_analysis},
  pdfauthor={Temitope},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}

\title{GSE211398\_race\_analysis}
\author{Temitope}
\date{2025-11-24}

\begin{document}
\maketitle

\subsection{R Markdown}\label{r-markdown}

This is dataset is gotten from the pubic dataset titled ``Comparative
gene expression profiling analysis of RNA-seq data for pancreatic
adenocarcinoma specimens obtained from Black and White patients''.
\url{https://pubmed.ncbi.nlm.nih.gov/36812168/}

This analysis seek to answer the question: does the tumour effect differ
by race?

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# 1. Load required libraries}
\FunctionTok{library}\NormalTok{(DESeq2)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Loading required package: S4Vectors
\end{verbatim}

\begin{verbatim}
## Loading required package: stats4
\end{verbatim}

\begin{verbatim}
## Loading required package: BiocGenerics
\end{verbatim}

\begin{verbatim}
## Loading required package: generics
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'generics'
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:base':
## 
##     as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,
##     setequal, union
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'BiocGenerics'
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:stats':
## 
##     IQR, mad, sd, var, xtabs
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:base':
## 
##     anyDuplicated, aperm, append, as.data.frame, basename, cbind,
##     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
##     get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,
##     unsplit, which.max, which.min
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'S4Vectors'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:utils':
## 
##     findMatches
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:base':
## 
##     expand.grid, I, unname
\end{verbatim}

\begin{verbatim}
## Loading required package: IRanges
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'IRanges'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:grDevices':
## 
##     windows
\end{verbatim}

\begin{verbatim}
## Loading required package: GenomicRanges
\end{verbatim}

\begin{verbatim}
## Loading required package: GenomeInfoDb
\end{verbatim}

\begin{verbatim}
## Loading required package: SummarizedExperiment
\end{verbatim}

\begin{verbatim}
## Loading required package: MatrixGenerics
\end{verbatim}

\begin{verbatim}
## Loading required package: matrixStats
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'MatrixGenerics'
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:matrixStats':
## 
##     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
##     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
##     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
##     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
##     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
##     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
##     colWeightedMeans, colWeightedMedians, colWeightedSds,
##     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
##     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
##     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
##     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
##     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
##     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
##     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
##     rowWeightedSds, rowWeightedVars
\end{verbatim}

\begin{verbatim}
## Loading required package: Biobase
\end{verbatim}

\begin{verbatim}
## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     'browseVignettes()'. To cite Bioconductor, see
##     'citation("Biobase")', and for packages 'citation("pkgname")'.
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'Biobase'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:MatrixGenerics':
## 
##     rowMedians
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:matrixStats':
## 
##     anyMissing, rowMedians
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(ggplot2)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning: package 'ggplot2' was built under R version 4.5.2
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(pheatmap)}
\FunctionTok{library}\NormalTok{(dplyr)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## Attaching package: 'dplyr'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:Biobase':
## 
##     combine
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:matrixStats':
## 
##     count
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:GenomicRanges':
## 
##     intersect, setdiff, union
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:GenomeInfoDb':
## 
##     intersect
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:IRanges':
## 
##     collapse, desc, intersect, setdiff, slice, union
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:S4Vectors':
## 
##     first, intersect, rename, setdiff, setequal, union
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:BiocGenerics':
## 
##     combine, intersect, setdiff, setequal, union
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:generics':
## 
##     explain
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:stats':
## 
##     filter, lag
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:base':
## 
##     intersect, setdiff, setequal, union
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(ggrepel)}
\FunctionTok{library}\NormalTok{(AnnotationDbi)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## Attaching package: 'AnnotationDbi'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:dplyr':
## 
##     select
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(org.Hs.eg.db)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# 2. Read count data and convert gene IDs to symbols}
\NormalTok{count\_data }\OtherTok{\textless{}{-}} \FunctionTok{read.delim}\NormalTok{(}\StringTok{"GSE211398\_raw\_counts\_GRCh38.p13\_NCBI.tsv"}\NormalTok{, }\AttributeTok{row.names =} \DecValTok{1}\NormalTok{)}
\NormalTok{count\_data }\OtherTok{\textless{}{-}}\NormalTok{ count\_data[}\FunctionTok{rowSums}\NormalTok{(count\_data) }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{, ]}

\CommentTok{\# Convert gene IDs to symbols}
\NormalTok{convert\_gene\_ids }\OtherTok{\textless{}{-}} \ControlFlowTok{function}\NormalTok{(gene\_ids) \{}
  \CommentTok{\# Remove version numbers if present}
\NormalTok{  clean\_ids }\OtherTok{\textless{}{-}} \FunctionTok{gsub}\NormalTok{(}\StringTok{"}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{..*"}\NormalTok{, }\StringTok{""}\NormalTok{, gene\_ids)}
  
  \CommentTok{\# Map to gene symbols}
\NormalTok{  symbols }\OtherTok{\textless{}{-}} \FunctionTok{mapIds}\NormalTok{(org.Hs.eg.db,}
                    \AttributeTok{keys =}\NormalTok{ clean\_ids,}
                    \AttributeTok{column =} \StringTok{"SYMBOL"}\NormalTok{,}
                    \AttributeTok{keytype =} \StringTok{"ENTREZID"}\NormalTok{,}
                    \AttributeTok{multiVals =} \StringTok{"first"}\NormalTok{)}
  \FunctionTok{return}\NormalTok{(symbols)}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Apply conversion}
\NormalTok{gene\_symbols }\OtherTok{\textless{}{-}} \FunctionTok{convert\_gene\_ids}\NormalTok{(}\FunctionTok{rownames}\NormalTok{(count\_data))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 'select()' returned 1:1 mapping between keys and columns
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Create a data frame to handle duplicates properly}
\NormalTok{gene\_mapping }\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(}
  \AttributeTok{original\_id =} \FunctionTok{rownames}\NormalTok{(count\_data),}
  \AttributeTok{symbol =}\NormalTok{ gene\_symbols,}
  \AttributeTok{stringsAsFactors =} \ConstantTok{FALSE}
\NormalTok{)}

\CommentTok{\# Remove rows where symbol conversion failed}
\NormalTok{gene\_mapping }\OtherTok{\textless{}{-}}\NormalTok{ gene\_mapping[}\SpecialCharTok{!}\FunctionTok{is.na}\NormalTok{(gene\_mapping}\SpecialCharTok{$}\NormalTok{symbol), ]}

\CommentTok{\# Calculate mean expression for each gene}
\NormalTok{mean\_expression }\OtherTok{\textless{}{-}} \FunctionTok{rowMeans}\NormalTok{(count\_data)}

\CommentTok{\# Add mean expression to mapping}
\NormalTok{gene\_mapping}\SpecialCharTok{$}\NormalTok{mean\_expr }\OtherTok{\textless{}{-}}\NormalTok{ mean\_expression[gene\_mapping}\SpecialCharTok{$}\NormalTok{original\_id]}

\CommentTok{\# For duplicate symbols, keep the one with highest mean expression}
\NormalTok{gene\_mapping }\OtherTok{\textless{}{-}}\NormalTok{ gene\_mapping }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{group\_by}\NormalTok{(symbol) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{arrange}\NormalTok{(}\FunctionTok{desc}\NormalTok{(mean\_expr)) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{slice}\NormalTok{(}\DecValTok{1}\NormalTok{) }\SpecialCharTok{\%\textgreater{}\%}  \CommentTok{\# Keep first row (highest expression) for each symbol}
  \FunctionTok{ungroup}\NormalTok{()}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Create new count matrix with unique gene symbols}
\NormalTok{count\_data\_symbols }\OtherTok{\textless{}{-}}\NormalTok{ count\_data[gene\_mapping}\SpecialCharTok{$}\NormalTok{original\_id, ]}
\FunctionTok{rownames}\NormalTok{(count\_data\_symbols) }\OtherTok{\textless{}{-}}\NormalTok{ gene\_mapping}\SpecialCharTok{$}\NormalTok{symbol}

\FunctionTok{print}\NormalTok{(}\FunctionTok{paste}\NormalTok{(}\StringTok{"Original genes:"}\NormalTok{, }\FunctionTok{nrow}\NormalTok{(count\_data)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "Original genes: 30384"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{print}\NormalTok{(}\FunctionTok{paste}\NormalTok{(}\StringTok{"Genes with unique symbols:"}\NormalTok{, }\FunctionTok{nrow}\NormalTok{(count\_data\_symbols)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "Genes with unique symbols: 29239"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{print}\NormalTok{(}\StringTok{"Sample of gene symbols:"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "Sample of gene symbols:"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{print}\NormalTok{(}\FunctionTok{head}\NormalTok{(}\FunctionTok{rownames}\NormalTok{(count\_data\_symbols), }\DecValTok{20}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##  [1] "A1BG"     "A1BG-AS1" "A1CF"     "A2M"      "A2M-AS1"  "A2ML1"   
##  [7] "A2MP1"    "A3GALT2"  "A4GALT"   "A4GNT"    "AAAS"     "AACS"    
## [13] "AACSP1"   "AADAC"    "AADACL2"  "AADACL3"  "AADACL4"  "AADACP1" 
## [19] "AADAT"    "AAGAB"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Check if we have common pancreatic cancer genes}
\NormalTok{pancreatic\_genes }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}\StringTok{"KRAS"}\NormalTok{, }\StringTok{"TP53"}\NormalTok{, }\StringTok{"CDKN2A"}\NormalTok{, }\StringTok{"SMAD4"}\NormalTok{, }\StringTok{"TSPAN8"}\NormalTok{, }\StringTok{"AGR2"}\NormalTok{, }\StringTok{"POSTN"}\NormalTok{, }\StringTok{"TFF1"}\NormalTok{, }\StringTok{"CP"}\NormalTok{)}

\NormalTok{found\_genes }\OtherTok{\textless{}{-}}\NormalTok{ pancreatic\_genes[pancreatic\_genes }\SpecialCharTok{\%in\%} \FunctionTok{rownames}\NormalTok{(count\_data\_symbols)]}

\FunctionTok{print}\NormalTok{(}\FunctionTok{paste}\NormalTok{(}\StringTok{"Found pancreatic cancer genes:"}\NormalTok{, }\FunctionTok{paste}\NormalTok{(found\_genes, }\AttributeTok{collapse =} \StringTok{", "}\NormalTok{)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "Found pancreatic cancer genes: KRAS, TP53, CDKN2A, SMAD4, TSPAN8, AGR2, POSTN, TFF1, CP"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# 3. EXTRACT METADATA FROM SERIES\_MATRIX FILE}
\NormalTok{series\_matrix }\OtherTok{\textless{}{-}} \FunctionTok{readLines}\NormalTok{(}\StringTok{"GSE211398\_series\_matrix.txt"}\NormalTok{)}

\CommentTok{\# Extract sample titles}
\NormalTok{sample\_title\_line }\OtherTok{\textless{}{-}} \FunctionTok{grep}\NormalTok{(}\StringTok{"!Sample\_title"}\NormalTok{, series\_matrix, }\AttributeTok{value =} \ConstantTok{TRUE}\NormalTok{)}
\NormalTok{sample\_titles }\OtherTok{\textless{}{-}} \FunctionTok{unlist}\NormalTok{(}\FunctionTok{strsplit}\NormalTok{(}\FunctionTok{gsub}\NormalTok{(}\StringTok{\textquotesingle{}!Sample\_title}\SpecialCharTok{\textbackslash{}t}\StringTok{"\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}\textquotesingle{}}\NormalTok{, sample\_title\_line), }\StringTok{\textquotesingle{}}\SpecialCharTok{\textbackslash{}t}\StringTok{\textquotesingle{}}\NormalTok{))}
\NormalTok{sample\_titles }\OtherTok{\textless{}{-}} \FunctionTok{gsub}\NormalTok{(}\StringTok{\textquotesingle{}"\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}\textquotesingle{}}\NormalTok{, sample\_titles)}

\CommentTok{\# Extract sample GEO accessions  }
\NormalTok{sample\_geo\_line }\OtherTok{\textless{}{-}} \FunctionTok{grep}\NormalTok{(}\StringTok{"!Sample\_geo\_accession"}\NormalTok{, series\_matrix, }\AttributeTok{value =} \ConstantTok{TRUE}\NormalTok{)}
\NormalTok{sample\_geo }\OtherTok{\textless{}{-}} \FunctionTok{unlist}\NormalTok{(}\FunctionTok{strsplit}\NormalTok{(}\FunctionTok{gsub}\NormalTok{(}\StringTok{\textquotesingle{}!Sample\_geo\_accession}\SpecialCharTok{\textbackslash{}t}\StringTok{"\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}\textquotesingle{}}\NormalTok{, sample\_geo\_line), }\StringTok{\textquotesingle{}}\SpecialCharTok{\textbackslash{}t}\StringTok{\textquotesingle{}}\NormalTok{))}
\NormalTok{sample\_geo }\OtherTok{\textless{}{-}} \FunctionTok{gsub}\NormalTok{(}\StringTok{\textquotesingle{}"\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}\textquotesingle{}}\NormalTok{, sample\_geo)}

\CommentTok{\# Extract characteristics}
\NormalTok{characteristics\_lines }\OtherTok{\textless{}{-}} \FunctionTok{grep}\NormalTok{(}\StringTok{"!Sample\_characteristics\_ch1"}\NormalTok{, series\_matrix, }\AttributeTok{value =} \ConstantTok{TRUE}\NormalTok{)}

\CommentTok{\# Parse characteristics}
\NormalTok{extract\_characteristics }\OtherTok{\textless{}{-}} \ControlFlowTok{function}\NormalTok{(line) \{}
\NormalTok{  content }\OtherTok{\textless{}{-}} \FunctionTok{gsub}\NormalTok{(}\StringTok{\textquotesingle{}!Sample\_characteristics\_ch1}\SpecialCharTok{\textbackslash{}t}\StringTok{"\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}\textquotesingle{}}\NormalTok{, line)}
\NormalTok{  content }\OtherTok{\textless{}{-}} \FunctionTok{gsub}\NormalTok{(}\StringTok{\textquotesingle{}"\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}\textquotesingle{}}\NormalTok{, content)}
  \FunctionTok{unlist}\NormalTok{(}\FunctionTok{strsplit}\NormalTok{(content, }\StringTok{\textquotesingle{}}\SpecialCharTok{\textbackslash{}t}\StringTok{\textquotesingle{}}\NormalTok{))}
\NormalTok{\}}

\NormalTok{tissue\_info }\OtherTok{\textless{}{-}} \FunctionTok{extract\_characteristics}\NormalTok{(characteristics\_lines[}\DecValTok{1}\NormalTok{])}
\NormalTok{race\_info }\OtherTok{\textless{}{-}} \FunctionTok{extract\_characteristics}\NormalTok{(characteristics\_lines[}\DecValTok{2}\NormalTok{])}
\NormalTok{gender\_info }\OtherTok{\textless{}{-}} \FunctionTok{extract\_characteristics}\NormalTok{(characteristics\_lines[}\DecValTok{3}\NormalTok{])}
\NormalTok{age\_info }\OtherTok{\textless{}{-}} \FunctionTok{extract\_characteristics}\NormalTok{(characteristics\_lines[}\DecValTok{4}\NormalTok{])}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Clean and create metadata}
\NormalTok{metadata }\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(}
  \AttributeTok{sample\_id =}\NormalTok{ sample\_geo,}
  \AttributeTok{tissue =} \FunctionTok{ifelse}\NormalTok{(}\FunctionTok{grepl}\NormalTok{(}\StringTok{"adenocarcinoma"}\NormalTok{, tissue\_info), }\StringTok{"Tumor"}\NormalTok{, }\StringTok{"NonTumor"}\NormalTok{),}
  \AttributeTok{race =} \FunctionTok{gsub}\NormalTok{(}\StringTok{"race: "}\NormalTok{, }\StringTok{""}\NormalTok{, race\_info),}
  \AttributeTok{gender =} \FunctionTok{gsub}\NormalTok{(}\StringTok{"gender: "}\NormalTok{, }\StringTok{""}\NormalTok{, gender\_info),}
  \AttributeTok{age =} \FunctionTok{as.numeric}\NormalTok{(}\FunctionTok{gsub}\NormalTok{(}\StringTok{"age: "}\NormalTok{, }\StringTok{""}\NormalTok{, age\_info)),}
  \AttributeTok{stringsAsFactors =} \ConstantTok{TRUE}
\NormalTok{)}
\FunctionTok{rownames}\NormalTok{(metadata) }\OtherTok{\textless{}{-}}\NormalTok{ metadata}\SpecialCharTok{$}\NormalTok{sample\_id}

\CommentTok{\# Reorder metadata to match count data}
\NormalTok{metadata }\OtherTok{\textless{}{-}}\NormalTok{ metadata[}\FunctionTok{colnames}\NormalTok{(count\_data\_symbols), ]}

\CommentTok{\# Verify metadata}
\FunctionTok{print}\NormalTok{(}\StringTok{"Sample distribution:"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "Sample distribution:"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{print}\NormalTok{(}\FunctionTok{table}\NormalTok{(metadata}\SpecialCharTok{$}\NormalTok{tissue, metadata}\SpecialCharTok{$}\NormalTok{race))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##           
##            Black White
##   NonTumor     3     9
##   Tumor        5    11
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{head}\NormalTok{(metadata)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##             sample_id tissue  race gender  age
## GSM6469488 GSM6469488  Tumor White   Male 72.8
## GSM6469489 GSM6469489  Tumor White   Male 78.5
## GSM6469490 GSM6469490  Tumor White   Male 76.8
## GSM6469491 GSM6469491  Tumor White Female 56.2
## GSM6469492 GSM6469492  Tumor Black Female 73.4
## GSM6469493 GSM6469493  Tumor White   Male 67.2
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Check sample distribution}
\NormalTok{sample\_summary }\OtherTok{\textless{}{-}}\NormalTok{ metadata }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{group\_by}\NormalTok{(tissue, race) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{summarise}\NormalTok{(}
    \AttributeTok{n =} \FunctionTok{n}\NormalTok{(),}
    \AttributeTok{mean\_age =} \FunctionTok{mean}\NormalTok{(age, }\AttributeTok{na.rm =} \ConstantTok{TRUE}\NormalTok{),}
    \AttributeTok{sd\_age =} \FunctionTok{sd}\NormalTok{(age, }\AttributeTok{na.rm =} \ConstantTok{TRUE}\NormalTok{),}
    \AttributeTok{.groups =} \StringTok{\textquotesingle{}drop\textquotesingle{}}
\NormalTok{  )}
\FunctionTok{print}\NormalTok{(sample\_summary)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## # A tibble: 4 x 5
##   tissue   race      n mean_age sd_age
##   <fct>    <fct> <int>    <dbl>  <dbl>
## 1 NonTumor Black     3     39.3  18.1 
## 2 NonTumor White     9     47.7  11.7 
## 3 Tumor    Black     5     66.8  11.2 
## 4 Tumor    White    11     67.2   7.99
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# 4. Create DESeq2 object}
\NormalTok{dds }\OtherTok{\textless{}{-}} \FunctionTok{DESeqDataSetFromMatrix}\NormalTok{(}
  \AttributeTok{countData =}\NormalTok{ count\_data\_symbols,}
  \AttributeTok{colData =}\NormalTok{ metadata,}
  \AttributeTok{design =} \SpecialCharTok{\textasciitilde{}}\NormalTok{ tissue }\SpecialCharTok{+}\NormalTok{ race }\SpecialCharTok{+}\NormalTok{ tissue}\SpecialCharTok{:}\NormalTok{race}
\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# 5. Run DESeq2}
\NormalTok{dds }\OtherTok{\textless{}{-}} \FunctionTok{DESeq}\NormalTok{(dds)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## estimating size factors
\end{verbatim}

\begin{verbatim}
## estimating dispersions
\end{verbatim}

\begin{verbatim}
## gene-wise dispersion estimates
\end{verbatim}

\begin{verbatim}
## mean-dispersion relationship
\end{verbatim}

\begin{verbatim}
## final dispersion estimates
\end{verbatim}

\begin{verbatim}
## fitting model and testing
\end{verbatim}

\begin{verbatim}
## -- replacing outliers and refitting for 541 genes
## -- DESeq argument 'minReplicatesForReplace' = 7 
## -- original counts are preserved in counts(dds)
\end{verbatim}

\begin{verbatim}
## estimating dispersions
\end{verbatim}

\begin{verbatim}
## fitting model and testing
\end{verbatim}

\subsubsection{6. ALL KEY COMPARISONS}\label{all-key-comparisons}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\# 6a. Tumor vs Non{-}Tumor (across all races)}
\NormalTok{res\_tumor\_vs\_normal }\OtherTok{\textless{}{-}} \FunctionTok{results}\NormalTok{(dds, }\AttributeTok{name =} \StringTok{"tissue\_Tumor\_vs\_NonTumor"}\NormalTok{)}
\NormalTok{sig\_tumor }\OtherTok{\textless{}{-}}\NormalTok{ res\_tumor\_vs\_normal[res\_tumor\_vs\_normal}\SpecialCharTok{$}\NormalTok{padj }\SpecialCharTok{\textless{}} \FloatTok{0.05} \SpecialCharTok{\&} \SpecialCharTok{!}\FunctionTok{is.na}\NormalTok{(res\_tumor\_vs\_normal}\SpecialCharTok{$}\NormalTok{padj), ]}
\NormalTok{sig\_tumor }\OtherTok{\textless{}{-}}\NormalTok{ sig\_tumor[}\FunctionTok{order}\NormalTok{(sig\_tumor}\SpecialCharTok{$}\NormalTok{padj), ]}

\FunctionTok{print}\NormalTok{(}\FunctionTok{paste}\NormalTok{(}\StringTok{"Number of DEGs between tumor and non{-}tumor:"}\NormalTok{, }\FunctionTok{nrow}\NormalTok{(sig\_tumor)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "Number of DEGs between tumor and non-tumor: 2582"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{print}\NormalTok{(}\StringTok{"Top 10 tumor vs non{-}tumor genes:"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "Top 10 tumor vs non-tumor genes:"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{top\_tumor\_genes }\OtherTok{\textless{}{-}} \FunctionTok{as.data.frame}\NormalTok{(}\FunctionTok{head}\NormalTok{(sig\_tumor, }\DecValTok{10}\NormalTok{))}
\NormalTok{top\_tumor\_genes}\SpecialCharTok{$}\NormalTok{gene\_symbol }\OtherTok{\textless{}{-}} \FunctionTok{rownames}\NormalTok{(top\_tumor\_genes)}
\FunctionTok{print}\NormalTok{(top\_tumor\_genes[, }\FunctionTok{c}\NormalTok{(}\StringTok{"gene\_symbol"}\NormalTok{, }\StringTok{"log2FoldChange"}\NormalTok{, }\StringTok{"padj"}\NormalTok{)])}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##           gene_symbol log2FoldChange         padj
## IGHV3-15     IGHV3-15       21.85430 4.267674e-24
## IGHV3-21     IGHV3-21       20.79325 1.773988e-22
## MUC17           MUC17       27.11863 1.773988e-22
## IGKV1D-33   IGKV1D-33       22.62090 7.880731e-21
## KRT16           KRT16       23.58299 8.593489e-20
## UTY               UTY       25.21620 8.593489e-20
## IGLV2-18     IGLV2-18       20.48105 3.177769e-19
## P2RY10         P2RY10       19.06878 3.210142e-19
## FCRL5           FCRL5       19.63746 5.536206e-19
## MUC2             MUC2       22.33241 1.252835e-18
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\# 6b. Race effect (across all tissues)}
\NormalTok{res\_race }\OtherTok{\textless{}{-}} \FunctionTok{results}\NormalTok{(dds, }\AttributeTok{name =} \StringTok{"race\_White\_vs\_Black"}\NormalTok{)}
\NormalTok{sig\_race }\OtherTok{\textless{}{-}}\NormalTok{ res\_race[res\_race}\SpecialCharTok{$}\NormalTok{padj }\SpecialCharTok{\textless{}} \FloatTok{0.05} \SpecialCharTok{\&} \SpecialCharTok{!}\FunctionTok{is.na}\NormalTok{(res\_race}\SpecialCharTok{$}\NormalTok{padj), ]}
\NormalTok{sig\_race }\OtherTok{\textless{}{-}}\NormalTok{ sig\_race[}\FunctionTok{order}\NormalTok{(sig\_race}\SpecialCharTok{$}\NormalTok{padj), ]}


\FunctionTok{print}\NormalTok{(}\FunctionTok{paste}\NormalTok{(}\StringTok{"Number of DEGs between races:"}\NormalTok{, }\FunctionTok{nrow}\NormalTok{(sig\_race)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "Number of DEGs between races: 334"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{print}\NormalTok{(}\StringTok{"Top 10 race{-}associated genes:"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "Top 10 race-associated genes:"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{top\_race\_genes }\OtherTok{\textless{}{-}} \FunctionTok{as.data.frame}\NormalTok{(}\FunctionTok{head}\NormalTok{(sig\_race, }\DecValTok{20}\NormalTok{))}
\NormalTok{top\_race\_genes}\SpecialCharTok{$}\NormalTok{gene\_symbol }\OtherTok{\textless{}{-}} \FunctionTok{rownames}\NormalTok{(top\_race\_genes)}
\FunctionTok{print}\NormalTok{(top\_race\_genes[, }\FunctionTok{c}\NormalTok{(}\StringTok{"gene\_symbol"}\NormalTok{, }\StringTok{"log2FoldChange"}\NormalTok{, }\StringTok{"padj"}\NormalTok{)])}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##               gene_symbol log2FoldChange         padj
## IGHV3-15         IGHV3-15       24.05654 1.191766e-33
## IGHV3-21         IGHV3-21       21.66123 1.167015e-27
## FCRL5               FCRL5       21.05130 1.521716e-24
## IGKV1D-33       IGKV1D-33       22.83885 2.002550e-24
## IGKV1-16         IGKV1-16       22.88549 8.773786e-23
## UTY                   UTY       25.08795 8.773786e-23
## ADGRG7             ADGRG7       21.57027 5.409480e-22
## P2RY10             P2RY10       19.20953 5.409480e-22
## IGLV2-18         IGLV2-18       20.33955 1.126361e-21
## IGHV4-4           IGHV4-4       21.61821 2.155190e-21
## IGKJ5               IGKJ5       21.71431 3.504131e-21
## MUC2                 MUC2       21.31686 1.711034e-19
## IGHV1OR15-1   IGHV1OR15-1       20.33627 3.115665e-19
## LOC102724760 LOC102724760       20.22679 6.187598e-18
## PKP1                 PKP1       20.17533 3.554007e-16
## BTLA                 BTLA       19.58205 6.866188e-16
## KRT16               KRT16       19.76065 6.866188e-16
## MUC17               MUC17       20.94462 1.744367e-15
## TIMD4               TIMD4       20.07639 1.861585e-12
## ELK2AP             ELK2AP       19.54860 6.416125e-12
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\# 6c. INTERACTION: Does tumor effect differ by race? }
\NormalTok{res\_interaction }\OtherTok{\textless{}{-}} \FunctionTok{results}\NormalTok{(dds, }\AttributeTok{name =} \StringTok{"tissueTumor.raceWhite"}\NormalTok{)}
\NormalTok{sig\_interaction }\OtherTok{\textless{}{-}}\NormalTok{ res\_interaction[res\_interaction}\SpecialCharTok{$}\NormalTok{padj }\SpecialCharTok{\textless{}} \FloatTok{0.05} \SpecialCharTok{\&} \SpecialCharTok{!}\FunctionTok{is.na}\NormalTok{(res\_interaction}\SpecialCharTok{$}\NormalTok{padj), ]}
\NormalTok{sig\_interaction }\OtherTok{\textless{}{-}}\NormalTok{ sig\_interaction[}\FunctionTok{order}\NormalTok{(sig\_interaction}\SpecialCharTok{$}\NormalTok{padj), ]}


\FunctionTok{print}\NormalTok{(}\FunctionTok{paste}\NormalTok{(}\StringTok{"Number of genes where tumor effect differs by race:"}\NormalTok{, }\FunctionTok{nrow}\NormalTok{(sig\_interaction)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "Number of genes where tumor effect differs by race: 123"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{print}\NormalTok{(}\StringTok{"Top 10 interaction genes:"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "Top 10 interaction genes:"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{top\_interaction\_genes }\OtherTok{\textless{}{-}} \FunctionTok{as.data.frame}\NormalTok{(}\FunctionTok{head}\NormalTok{(sig\_interaction, }\DecValTok{10}\NormalTok{))}
\NormalTok{top\_interaction\_genes}\SpecialCharTok{$}\NormalTok{gene\_symbol }\OtherTok{\textless{}{-}} \FunctionTok{rownames}\NormalTok{(top\_interaction\_genes)}
\FunctionTok{print}\NormalTok{(top\_interaction\_genes[, }\FunctionTok{c}\NormalTok{(}\StringTok{"gene\_symbol"}\NormalTok{, }\StringTok{"log2FoldChange"}\NormalTok{, }\StringTok{"padj"}\NormalTok{)])}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##           gene_symbol log2FoldChange         padj
## IGHV3-15     IGHV3-15      -22.06786 2.837554e-19
## UTY               UTY      -26.09945 1.493746e-15
## FCRL5           FCRL5      -19.91086 6.122298e-15
## IGHV3-21     IGHV3-21      -19.21337 6.122298e-15
## IGKV1D-33   IGKV1D-33      -21.26518 6.174927e-14
## P2RY10         P2RY10      -18.39733 8.516423e-14
## ADGRG7         ADGRG7      -20.32747 9.860211e-13
## MUC2             MUC2      -21.02201 2.983747e-12
## IGLV2-18     IGLV2-18      -18.65143 4.010054e-12
## IGKV1-16     IGKV1-16      -20.52867 6.380199e-12
\end{verbatim}

\subsubsection{VISUALIZING ALL
COMPARISONS}\label{visualizing-all-comparisons}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\# 7a. Volcano plot for Tumor vs Non{-}Tumor}
\NormalTok{volcano\_tumor }\OtherTok{\textless{}{-}} \FunctionTok{as.data.frame}\NormalTok{(res\_tumor\_vs\_normal)}
\NormalTok{volcano\_tumor}\SpecialCharTok{$}\NormalTok{gene }\OtherTok{\textless{}{-}} \FunctionTok{rownames}\NormalTok{(volcano\_tumor)}
\NormalTok{volcano\_tumor}\SpecialCharTok{$}\NormalTok{significant }\OtherTok{\textless{}{-}}\NormalTok{ volcano\_tumor}\SpecialCharTok{$}\NormalTok{padj }\SpecialCharTok{\textless{}} \FloatTok{0.05} \SpecialCharTok{\&} \FunctionTok{abs}\NormalTok{(volcano\_tumor}\SpecialCharTok{$}\NormalTok{log2FoldChange) }\SpecialCharTok{\textgreater{}} \DecValTok{1}
\NormalTok{volcano\_tumor}\SpecialCharTok{$}\NormalTok{significant[}\FunctionTok{is.na}\NormalTok{(volcano\_tumor}\SpecialCharTok{$}\NormalTok{significant)] }\OtherTok{\textless{}{-}} \ConstantTok{FALSE}

\CommentTok{\# Label top genes}
\NormalTok{volcano\_tumor}\SpecialCharTok{$}\NormalTok{label }\OtherTok{\textless{}{-}} \FunctionTok{ifelse}\NormalTok{(volcano\_tumor}\SpecialCharTok{$}\NormalTok{padj }\SpecialCharTok{\textless{}} \FloatTok{0.001} \SpecialCharTok{\&} \FunctionTok{abs}\NormalTok{(volcano\_tumor}\SpecialCharTok{$}\NormalTok{log2FoldChange) }\SpecialCharTok{\textgreater{}} \DecValTok{2}\NormalTok{, }
\NormalTok{                              volcano\_tumor}\SpecialCharTok{$}\NormalTok{gene, }\StringTok{""}\NormalTok{)}

\FunctionTok{ggplot}\NormalTok{(volcano\_tumor, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ log2FoldChange, }\AttributeTok{y =} \SpecialCharTok{{-}}\FunctionTok{log10}\NormalTok{(pvalue))) }\SpecialCharTok{+}
  \FunctionTok{geom\_point}\NormalTok{(}\FunctionTok{aes}\NormalTok{(}\AttributeTok{color =}\NormalTok{ significant), }\AttributeTok{alpha =} \FloatTok{0.6}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_text\_repel}\NormalTok{(}\FunctionTok{aes}\NormalTok{(}\AttributeTok{label =}\NormalTok{ label), }\AttributeTok{size =} \DecValTok{3}\NormalTok{, }\AttributeTok{max.overlaps =} \DecValTok{10}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{scale\_color\_manual}\NormalTok{(}\AttributeTok{values =} \FunctionTok{c}\NormalTok{(}\StringTok{"grey"}\NormalTok{, }\StringTok{"red"}\NormalTok{)) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}\AttributeTok{title =} \StringTok{"Tumor vs Non{-}Tumor DEGs"}\NormalTok{,}
       \AttributeTok{x =} \StringTok{"Log2 Fold Change (Tumor/NonTumor)"}\NormalTok{,}
       \AttributeTok{y =} \StringTok{"{-}Log10 P{-}value"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{theme\_minimal}\NormalTok{() }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}\AttributeTok{legend.position =} \StringTok{"none"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning: Removed 168 rows containing missing values or values outside the scale range
## (`geom_point()`).
\end{verbatim}

\begin{verbatim}
## Warning: Removed 3891 rows containing missing values or values outside the scale range
## (`geom_text_repel()`).
\end{verbatim}

\begin{verbatim}
## Warning: ggrepel: 451 unlabeled data points (too many overlaps). Consider
## increasing max.overlaps
\end{verbatim}

\pandocbounded{\includegraphics[keepaspectratio]{GSE211398_DeseqAnalysis_files/figure-latex/unnamed-chunk-16-1.pdf}}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\# 7b. Volcano plot for Race effect}
\NormalTok{volcano\_race }\OtherTok{\textless{}{-}} \FunctionTok{as.data.frame}\NormalTok{(res\_race)}
\NormalTok{volcano\_race}\SpecialCharTok{$}\NormalTok{gene }\OtherTok{\textless{}{-}} \FunctionTok{rownames}\NormalTok{(volcano\_race)}
\NormalTok{volcano\_race}\SpecialCharTok{$}\NormalTok{significant }\OtherTok{\textless{}{-}}\NormalTok{ volcano\_race}\SpecialCharTok{$}\NormalTok{padj }\SpecialCharTok{\textless{}} \FloatTok{0.05} \SpecialCharTok{\&} \FunctionTok{abs}\NormalTok{(volcano\_race}\SpecialCharTok{$}\NormalTok{log2FoldChange) }\SpecialCharTok{\textgreater{}} \DecValTok{1}
\NormalTok{volcano\_race}\SpecialCharTok{$}\NormalTok{significant[}\FunctionTok{is.na}\NormalTok{(volcano\_race}\SpecialCharTok{$}\NormalTok{significant)] }\OtherTok{\textless{}{-}} \ConstantTok{FALSE}

\NormalTok{volcano\_race}\SpecialCharTok{$}\NormalTok{label }\OtherTok{\textless{}{-}} \FunctionTok{ifelse}\NormalTok{(volcano\_race}\SpecialCharTok{$}\NormalTok{padj }\SpecialCharTok{\textless{}} \FloatTok{0.001} \SpecialCharTok{\&} \FunctionTok{abs}\NormalTok{(volcano\_race}\SpecialCharTok{$}\NormalTok{log2FoldChange) }\SpecialCharTok{\textgreater{}} \DecValTok{2}\NormalTok{, }
\NormalTok{                             volcano\_race}\SpecialCharTok{$}\NormalTok{gene, }\StringTok{""}\NormalTok{)}

\FunctionTok{ggplot}\NormalTok{(volcano\_race, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ log2FoldChange, }\AttributeTok{y =} \SpecialCharTok{{-}}\FunctionTok{log10}\NormalTok{(pvalue))) }\SpecialCharTok{+}
  \FunctionTok{geom\_point}\NormalTok{(}\FunctionTok{aes}\NormalTok{(}\AttributeTok{color =}\NormalTok{ significant), }\AttributeTok{alpha =} \FloatTok{0.6}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_text\_repel}\NormalTok{(}\FunctionTok{aes}\NormalTok{(}\AttributeTok{label =}\NormalTok{ label), }\AttributeTok{size =} \DecValTok{3}\NormalTok{, }\AttributeTok{max.overlaps =} \DecValTok{10}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{scale\_color\_manual}\NormalTok{(}\AttributeTok{values =} \FunctionTok{c}\NormalTok{(}\StringTok{"grey"}\NormalTok{, }\StringTok{"blue"}\NormalTok{)) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}\AttributeTok{title =} \StringTok{"White vs Black DEGs"}\NormalTok{,}
       \AttributeTok{x =} \StringTok{"Log2 Fold Change (White/Black)"}\NormalTok{,}
       \AttributeTok{y =} \StringTok{"{-}Log10 P{-}value"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{theme\_minimal}\NormalTok{() }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}\AttributeTok{legend.position =} \StringTok{"none"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning: Removed 168 rows containing missing values or values outside the scale range
## (`geom_point()`).
\end{verbatim}

\begin{verbatim}
## Warning: Removed 1126 rows containing missing values or values outside the scale range
## (`geom_text_repel()`).
\end{verbatim}

\begin{verbatim}
## Warning: ggrepel: 57 unlabeled data points (too many overlaps). Consider
## increasing max.overlaps
\end{verbatim}

\pandocbounded{\includegraphics[keepaspectratio]{GSE211398_DeseqAnalysis_files/figure-latex/unnamed-chunk-17-1.pdf}}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\# 7c. Volcano plot for Interaction}
\NormalTok{volcano\_interaction }\OtherTok{\textless{}{-}} \FunctionTok{as.data.frame}\NormalTok{(res\_interaction)}
\NormalTok{volcano\_interaction}\SpecialCharTok{$}\NormalTok{gene }\OtherTok{\textless{}{-}} \FunctionTok{rownames}\NormalTok{(volcano\_interaction)}
\NormalTok{volcano\_interaction}\SpecialCharTok{$}\NormalTok{significant }\OtherTok{\textless{}{-}}\NormalTok{ volcano\_interaction}\SpecialCharTok{$}\NormalTok{padj }\SpecialCharTok{\textless{}} \FloatTok{0.05} \SpecialCharTok{\&} \FunctionTok{abs}\NormalTok{(volcano\_interaction}\SpecialCharTok{$}\NormalTok{log2FoldChange) }\SpecialCharTok{\textgreater{}} \DecValTok{1}
\NormalTok{volcano\_interaction}\SpecialCharTok{$}\NormalTok{significant[}\FunctionTok{is.na}\NormalTok{(volcano\_interaction}\SpecialCharTok{$}\NormalTok{significant)] }\OtherTok{\textless{}{-}} \ConstantTok{FALSE}

\NormalTok{volcano\_interaction}\SpecialCharTok{$}\NormalTok{label }\OtherTok{\textless{}{-}} \FunctionTok{ifelse}\NormalTok{(volcano\_interaction}\SpecialCharTok{$}\NormalTok{padj }\SpecialCharTok{\textless{}} \FloatTok{0.001} \SpecialCharTok{\&} \FunctionTok{abs}\NormalTok{(volcano\_interaction}\SpecialCharTok{$}\NormalTok{log2FoldChange) }\SpecialCharTok{\textgreater{}} \DecValTok{2}\NormalTok{, }
\NormalTok{                                    volcano\_interaction}\SpecialCharTok{$}\NormalTok{gene, }\StringTok{""}\NormalTok{)}

\FunctionTok{ggplot}\NormalTok{(volcano\_interaction, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ log2FoldChange, }\AttributeTok{y =} \SpecialCharTok{{-}}\FunctionTok{log10}\NormalTok{(pvalue))) }\SpecialCharTok{+}
  \FunctionTok{geom\_point}\NormalTok{(}\FunctionTok{aes}\NormalTok{(}\AttributeTok{color =}\NormalTok{ significant), }\AttributeTok{alpha =} \FloatTok{0.6}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_text\_repel}\NormalTok{(}\FunctionTok{aes}\NormalTok{(}\AttributeTok{label =}\NormalTok{ label), }\AttributeTok{size =} \DecValTok{3}\NormalTok{, }\AttributeTok{max.overlaps =} \DecValTok{10}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{scale\_color\_manual}\NormalTok{(}\AttributeTok{values =} \FunctionTok{c}\NormalTok{(}\StringTok{"grey"}\NormalTok{, }\StringTok{"purple"}\NormalTok{)) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}\AttributeTok{title =} \StringTok{"Interaction: Genes where tumor effect differs by race"}\NormalTok{,}
       \AttributeTok{x =} \StringTok{"Log2 Fold Change (Interaction)"}\NormalTok{,}
       \AttributeTok{y =} \StringTok{"{-}Log10 P{-}value"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{theme\_minimal}\NormalTok{() }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}\AttributeTok{legend.position =} \StringTok{"none"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning: Removed 168 rows containing missing values or values outside the scale range
## (`geom_point()`).
\end{verbatim}

\begin{verbatim}
## Warning: Removed 2242 rows containing missing values or values outside the scale range
## (`geom_text_repel()`).
\end{verbatim}

\begin{verbatim}
## Warning: ggrepel: 6 unlabeled data points (too many overlaps). Consider
## increasing max.overlaps
\end{verbatim}

\pandocbounded{\includegraphics[keepaspectratio]{GSE211398_DeseqAnalysis_files/figure-latex/unnamed-chunk-18-1.pdf}}

\subsubsection{8. HEATMAPS FOR EACH
COMPARISON}\label{heatmaps-for-each-comparison}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\# 8a. Heatmap for Tumor vs Non{-}Tumor DEGs}
\ControlFlowTok{if}\NormalTok{ (}\FunctionTok{nrow}\NormalTok{(sig\_tumor) }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{) \{}
\NormalTok{  vsd }\OtherTok{\textless{}{-}} \FunctionTok{vst}\NormalTok{(dds, }\AttributeTok{blind =} \ConstantTok{FALSE}\NormalTok{)}
\NormalTok{  top\_tumor\_genes }\OtherTok{\textless{}{-}} \FunctionTok{rownames}\NormalTok{(sig\_tumor)[}\DecValTok{1}\SpecialCharTok{:}\FunctionTok{min}\NormalTok{(}\DecValTok{30}\NormalTok{, }\FunctionTok{nrow}\NormalTok{(sig\_tumor))]}
\NormalTok{  mat\_tumor }\OtherTok{\textless{}{-}} \FunctionTok{assay}\NormalTok{(vsd)[top\_tumor\_genes, ]}
\NormalTok{  mat\_tumor }\OtherTok{\textless{}{-}} \FunctionTok{t}\NormalTok{(}\FunctionTok{scale}\NormalTok{(}\FunctionTok{t}\NormalTok{(mat\_tumor)))}
  
  \FunctionTok{pheatmap}\NormalTok{(mat\_tumor,}
           \AttributeTok{annotation\_col =}\NormalTok{ metadata[, }\FunctionTok{c}\NormalTok{(}\StringTok{"tissue"}\NormalTok{, }\StringTok{"race"}\NormalTok{)],}
           \AttributeTok{main =} \StringTok{"Top Tumor vs Non{-}Tumor DEGs"}\NormalTok{,}
           \AttributeTok{show\_rownames =} \ConstantTok{TRUE}\NormalTok{,}
           \AttributeTok{fontsize\_row =} \DecValTok{8}\NormalTok{)}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\pandocbounded{\includegraphics[keepaspectratio]{GSE211398_DeseqAnalysis_files/figure-latex/unnamed-chunk-19-1.pdf}}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\# 8b. Heatmap for Race DEGs}
\ControlFlowTok{if}\NormalTok{ (}\FunctionTok{nrow}\NormalTok{(sig\_race) }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{) \{}
\NormalTok{  top\_race\_genes }\OtherTok{\textless{}{-}} \FunctionTok{rownames}\NormalTok{(sig\_race)[}\DecValTok{1}\SpecialCharTok{:}\FunctionTok{min}\NormalTok{(}\DecValTok{30}\NormalTok{, }\FunctionTok{nrow}\NormalTok{(sig\_race))]}
\NormalTok{  mat\_race }\OtherTok{\textless{}{-}} \FunctionTok{assay}\NormalTok{(vsd)[top\_race\_genes, ]}
\NormalTok{  mat\_race }\OtherTok{\textless{}{-}} \FunctionTok{t}\NormalTok{(}\FunctionTok{scale}\NormalTok{(}\FunctionTok{t}\NormalTok{(mat\_race)))}
  
  \FunctionTok{pheatmap}\NormalTok{(mat\_race,}
           \AttributeTok{annotation\_col =}\NormalTok{ metadata[, }\FunctionTok{c}\NormalTok{(}\StringTok{"tissue"}\NormalTok{, }\StringTok{"race"}\NormalTok{)],}
           \AttributeTok{main =} \StringTok{"Top Race{-}Associated DEGs"}\NormalTok{,}
           \AttributeTok{show\_rownames =} \ConstantTok{TRUE}\NormalTok{,}
           \AttributeTok{fontsize\_row =} \DecValTok{8}\NormalTok{)}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\pandocbounded{\includegraphics[keepaspectratio]{GSE211398_DeseqAnalysis_files/figure-latex/unnamed-chunk-20-1.pdf}}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\# 8c. Heatmap for Interaction genes}
\ControlFlowTok{if}\NormalTok{ (}\FunctionTok{nrow}\NormalTok{(sig\_interaction) }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{) \{}
\NormalTok{  top\_interaction\_genes }\OtherTok{\textless{}{-}} \FunctionTok{rownames}\NormalTok{(sig\_interaction)[}\DecValTok{1}\SpecialCharTok{:}\FunctionTok{min}\NormalTok{(}\DecValTok{30}\NormalTok{, }\FunctionTok{nrow}\NormalTok{(sig\_interaction))]}
\NormalTok{  mat\_interaction }\OtherTok{\textless{}{-}} \FunctionTok{assay}\NormalTok{(vsd)[top\_interaction\_genes, ]}
\NormalTok{  mat\_interaction }\OtherTok{\textless{}{-}} \FunctionTok{t}\NormalTok{(}\FunctionTok{scale}\NormalTok{(}\FunctionTok{t}\NormalTok{(mat\_interaction)))}
  
  \FunctionTok{pheatmap}\NormalTok{(mat\_interaction,}
           \AttributeTok{annotation\_col =}\NormalTok{ metadata[, }\FunctionTok{c}\NormalTok{(}\StringTok{"tissue"}\NormalTok{, }\StringTok{"race"}\NormalTok{)],}
           \AttributeTok{main =} \StringTok{"Top Genes: Tumor Effect Differs by Race"}\NormalTok{,}
           \AttributeTok{show\_rownames =} \ConstantTok{TRUE}\NormalTok{,}
           \AttributeTok{fontsize\_row =} \DecValTok{8}\NormalTok{)}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\pandocbounded{\includegraphics[keepaspectratio]{GSE211398_DeseqAnalysis_files/figure-latex/unnamed-chunk-21-1.pdf}}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# 9. Checking if  SPECIFIC pancreatic GENES are expressed}
\NormalTok{genes\_to\_check }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}\StringTok{"TSPAN8"}\NormalTok{, }\StringTok{"AGR2"}\NormalTok{, }\StringTok{"POSTN"}\NormalTok{, }\StringTok{"TFF1"}\NormalTok{, }\StringTok{"CP"}\NormalTok{, }\StringTok{"KRAS"}\NormalTok{, }\StringTok{"TP53"}\NormalTok{, }\StringTok{"CDKN2A"}\NormalTok{, }\StringTok{"SMAD4"}\NormalTok{)}

\FunctionTok{print}\NormalTok{(}\StringTok{"=== SPECIFIC GENE ANALYSIS ==="}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "=== SPECIFIC GENE ANALYSIS ==="
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\ControlFlowTok{for}\NormalTok{ (gene }\ControlFlowTok{in}\NormalTok{ genes\_to\_check) \{}
  \ControlFlowTok{if}\NormalTok{ (gene }\SpecialCharTok{\%in\%} \FunctionTok{rownames}\NormalTok{(count\_data\_symbols)) \{}
    \CommentTok{\# Plot expression}
\NormalTok{    plot\_data }\OtherTok{\textless{}{-}} \FunctionTok{plotCounts}\NormalTok{(dds, }\AttributeTok{gene =}\NormalTok{ gene, }\AttributeTok{intgroup =} \FunctionTok{c}\NormalTok{(}\StringTok{"tissue"}\NormalTok{, }\StringTok{"race"}\NormalTok{), }\AttributeTok{returnData =} \ConstantTok{TRUE}\NormalTok{)}
    
\NormalTok{    p }\OtherTok{\textless{}{-}} \FunctionTok{ggplot}\NormalTok{(plot\_data, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ tissue, }\AttributeTok{y =}\NormalTok{ count, }\AttributeTok{color =}\NormalTok{ race, }\AttributeTok{shape =}\NormalTok{ race)) }\SpecialCharTok{+}
      \FunctionTok{geom\_point}\NormalTok{(}\AttributeTok{position =} \FunctionTok{position\_jitter}\NormalTok{(}\AttributeTok{width =} \FloatTok{0.2}\NormalTok{), }\AttributeTok{size =} \DecValTok{3}\NormalTok{) }\SpecialCharTok{+}
      \FunctionTok{labs}\NormalTok{(}\AttributeTok{title =} \FunctionTok{paste}\NormalTok{(}\StringTok{"Expression of"}\NormalTok{, gene),}
           \AttributeTok{x =} \StringTok{"Tissue"}\NormalTok{, }\AttributeTok{y =} \StringTok{"Normalized Counts"}\NormalTok{) }\SpecialCharTok{+}
      \FunctionTok{theme\_minimal}\NormalTok{()}
    \FunctionTok{print}\NormalTok{(p) }\CommentTok{\# Print results for this gene}
    \FunctionTok{cat}\NormalTok{(}\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{{-}{-}{-}"}\NormalTok{, gene, }\StringTok{"{-}{-}{-}}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
    \ControlFlowTok{if}\NormalTok{ (gene }\SpecialCharTok{\%in\%} \FunctionTok{rownames}\NormalTok{(res\_tumor\_vs\_normal)) \{}
\NormalTok{      tumor\_result }\OtherTok{\textless{}{-}}\NormalTok{ res\_tumor\_vs\_normal[gene, ]}
\NormalTok{      direction }\OtherTok{\textless{}{-}} \FunctionTok{ifelse}\NormalTok{(tumor\_result}\SpecialCharTok{$}\NormalTok{log2FoldChange }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{, }\StringTok{"UP in Tumor"}\NormalTok{, }\StringTok{"DOWN in Tumor"}\NormalTok{)}
      \FunctionTok{cat}\NormalTok{(}\StringTok{"Tumor vs Normal: FC ="}\NormalTok{, }\FunctionTok{round}\NormalTok{(}\DecValTok{2}\SpecialCharTok{\^{}}\NormalTok{tumor\_result}\SpecialCharTok{$}\NormalTok{log2FoldChange, }\DecValTok{2}\NormalTok{), }
          \StringTok{"| padj ="}\NormalTok{, }\FunctionTok{format}\NormalTok{(tumor\_result}\SpecialCharTok{$}\NormalTok{padj, }\AttributeTok{scientific =} \ConstantTok{TRUE}\NormalTok{), }\StringTok{"|"}\NormalTok{, direction, }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\NormalTok{    \}}
    
    \ControlFlowTok{if}\NormalTok{ (gene }\SpecialCharTok{\%in\%} \FunctionTok{rownames}\NormalTok{(res\_race)) \{}
\NormalTok{      race\_result }\OtherTok{\textless{}{-}}\NormalTok{ res\_race[gene, ]}
\NormalTok{      direction }\OtherTok{\textless{}{-}} \FunctionTok{ifelse}\NormalTok{(race\_result}\SpecialCharTok{$}\NormalTok{log2FoldChange }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{, }\StringTok{"UP in White"}\NormalTok{, }\StringTok{"UP in Black"}\NormalTok{)}
      \FunctionTok{cat}\NormalTok{(}\StringTok{"Race effect: FC ="}\NormalTok{, }\FunctionTok{round}\NormalTok{(}\DecValTok{2}\SpecialCharTok{\^{}}\NormalTok{race\_result}\SpecialCharTok{$}\NormalTok{log2FoldChange, }\DecValTok{2}\NormalTok{), }
          \StringTok{"| padj ="}\NormalTok{, }\FunctionTok{format}\NormalTok{(race\_result}\SpecialCharTok{$}\NormalTok{padj, }\AttributeTok{scientific =} \ConstantTok{TRUE}\NormalTok{), }\StringTok{"|"}\NormalTok{, direction, }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\NormalTok{    \}}
    
    \ControlFlowTok{if}\NormalTok{ (gene }\SpecialCharTok{\%in\%} \FunctionTok{rownames}\NormalTok{(res\_interaction)) \{}
\NormalTok{      interaction\_result }\OtherTok{\textless{}{-}}\NormalTok{ res\_interaction[gene, ]}
\NormalTok{      direction }\OtherTok{\textless{}{-}} \FunctionTok{ifelse}\NormalTok{(interaction\_result}\SpecialCharTok{$}\NormalTok{log2FoldChange }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{, }\StringTok{"Stronger in White"}\NormalTok{, }\StringTok{"Stronger in Black"}\NormalTok{)}
      \FunctionTok{cat}\NormalTok{(}\StringTok{"Interaction: FC ="}\NormalTok{, }\FunctionTok{round}\NormalTok{(interaction\_result}\SpecialCharTok{$}\NormalTok{log2FoldChange, }\DecValTok{2}\NormalTok{), }
          \StringTok{"| padj ="}\NormalTok{, }\FunctionTok{format}\NormalTok{(interaction\_result}\SpecialCharTok{$}\NormalTok{padj, }\AttributeTok{scientific =} \ConstantTok{TRUE}\NormalTok{), }\StringTok{"| Tumor effect"}\NormalTok{, direction, }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\NormalTok{    \}}
\NormalTok{  \} }\ControlFlowTok{else}\NormalTok{ \{}
    \FunctionTok{cat}\NormalTok{(}\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, gene, }\StringTok{"not found in dataset}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\NormalTok{  \}}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\pandocbounded{\includegraphics[keepaspectratio]{GSE211398_DeseqAnalysis_files/figure-latex/unnamed-chunk-22-1.pdf}}

\begin{verbatim}
## 
## --- TSPAN8 ---
## Tumor vs Normal: FC = 29.17 | padj = 3.386687e-04 | UP in Tumor 
## Race effect: FC = 1.38 | padj = 9.622917e-01 | UP in White 
## Interaction: FC = -2.29 | padj = 7.432812e-01 | Tumor effect Stronger in Black
\end{verbatim}

\pandocbounded{\includegraphics[keepaspectratio]{GSE211398_DeseqAnalysis_files/figure-latex/unnamed-chunk-22-2.pdf}}

\begin{verbatim}
## 
## --- AGR2 ---
## Tumor vs Normal: FC = 13 | padj = 1.834723e-02 | UP in Tumor 
## Race effect: FC = 0.6 | padj = 9.208789e-01 | UP in Black 
## Interaction: FC = 0.63 | padj = 9.969893e-01 | Tumor effect Stronger in White
\end{verbatim}

\pandocbounded{\includegraphics[keepaspectratio]{GSE211398_DeseqAnalysis_files/figure-latex/unnamed-chunk-22-3.pdf}}

\begin{verbatim}
## 
## --- POSTN ---
## Tumor vs Normal: FC = 176.42 | padj = 3.126186e-13 | UP in Tumor 
## Race effect: FC = 3.74 | padj = 3.47969e-01 | UP in White 
## Interaction: FC = -1.38 | padj = 9.079409e-01 | Tumor effect Stronger in Black
\end{verbatim}

\pandocbounded{\includegraphics[keepaspectratio]{GSE211398_DeseqAnalysis_files/figure-latex/unnamed-chunk-22-4.pdf}}

\begin{verbatim}
## 
## --- TFF1 ---
## Tumor vs Normal: FC = 7838.04 | padj = 2.657379e-09 | UP in Tumor 
## Race effect: FC = 213.15 | padj = 2.249158e-03 | UP in White 
## Interaction: FC = -8.3 | padj = 1.779011e-02 | Tumor effect Stronger in Black
\end{verbatim}

\pandocbounded{\includegraphics[keepaspectratio]{GSE211398_DeseqAnalysis_files/figure-latex/unnamed-chunk-22-5.pdf}}

\begin{verbatim}
## 
## --- CP ---
## Tumor vs Normal: FC = 11.49 | padj = 6.314581e-03 | UP in Tumor 
## Race effect: FC = 0.98 | padj = 9.97813e-01 | UP in Black 
## Interaction: FC = 2.07 | padj = 7.461765e-01 | Tumor effect Stronger in White
\end{verbatim}

\pandocbounded{\includegraphics[keepaspectratio]{GSE211398_DeseqAnalysis_files/figure-latex/unnamed-chunk-22-6.pdf}}

\begin{verbatim}
## 
## --- KRAS ---
## Tumor vs Normal: FC = 1.84 | padj = 2.110918e-01 | UP in Tumor 
## Race effect: FC = 0.81 | padj = 9.197895e-01 | UP in Black 
## Interaction: FC = 0.48 | padj = 9.867653e-01 | Tumor effect Stronger in White
\end{verbatim}

\pandocbounded{\includegraphics[keepaspectratio]{GSE211398_DeseqAnalysis_files/figure-latex/unnamed-chunk-22-7.pdf}}

\begin{verbatim}
## 
## --- TP53 ---
## Tumor vs Normal: FC = 1.14 | padj = 7.582079e-01 | UP in Tumor 
## Race effect: FC = 0.69 | padj = 5.938159e-01 | UP in Black 
## Interaction: FC = 0.47 | padj = 9.334379e-01 | Tumor effect Stronger in White
\end{verbatim}

\pandocbounded{\includegraphics[keepaspectratio]{GSE211398_DeseqAnalysis_files/figure-latex/unnamed-chunk-22-8.pdf}}

\begin{verbatim}
## 
## --- CDKN2A ---
## Tumor vs Normal: FC = 104.07 | padj = 1.095659e-04 | UP in Tumor 
## Race effect: FC = 98.37 | padj = 2.89949e-04 | UP in White 
## Interaction: FC = -6.03 | padj = 1.554318e-02 | Tumor effect Stronger in Black
\end{verbatim}

\pandocbounded{\includegraphics[keepaspectratio]{GSE211398_DeseqAnalysis_files/figure-latex/unnamed-chunk-22-9.pdf}}

\begin{verbatim}
## 
## --- SMAD4 ---
## Tumor vs Normal: FC = 0.76 | padj = 5.45444e-01 | DOWN in Tumor 
## Race effect: FC = 0.96 | padj = 9.912771e-01 | UP in Black 
## Interaction: FC = -0.17 | padj = 9.969893e-01 | Tumor effect Stronger in Black
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# 10. SUMMARY STATISTICS}
\FunctionTok{print}\NormalTok{(}\FunctionTok{paste}\NormalTok{(}\StringTok{"Tumor vs Non{-}Tumor DEGs:"}\NormalTok{, }\FunctionTok{nrow}\NormalTok{(sig\_tumor)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "Tumor vs Non-Tumor DEGs: 2582"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{print}\NormalTok{(}\FunctionTok{paste}\NormalTok{(}\StringTok{"Race{-}associated DEGs:"}\NormalTok{, }\FunctionTok{nrow}\NormalTok{(sig\_race)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "Race-associated DEGs: 334"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{print}\NormalTok{(}\FunctionTok{paste}\NormalTok{(}\StringTok{"Genes with different tumor effects by race:"}\NormalTok{, }\FunctionTok{nrow}\NormalTok{(sig\_interaction)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "Genes with different tumor effects by race: 123"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Save results with gene symbols}
\FunctionTok{write.csv}\NormalTok{(}\FunctionTok{as.data.frame}\NormalTok{(sig\_tumor), }\StringTok{"tumor\_vs\_normal\_deg.csv"}\NormalTok{)}
\FunctionTok{write.csv}\NormalTok{(}\FunctionTok{as.data.frame}\NormalTok{(sig\_race), }\StringTok{"race\_deg.csv"}\NormalTok{)}
\FunctionTok{write.csv}\NormalTok{(}\FunctionTok{as.data.frame}\NormalTok{(sig\_interaction), }\StringTok{"interaction\_genes.csv"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}


\end{document}
